src/App.tsx

```diff
@@ -1,15 +1,52 @@
-import React from 'react';
+import React, { useMemo, useState } from 'react';
 import './App.scss';
 import { peopleFromServer } from './data/people';
+import { Person } from './types/Person';
 
 export const App: React.FC = () => {
-  const { name, born, died } = peopleFromServer[0];
+  const [selectedPerson, setSelectedPerson] = useState<Person | null>(null);
+  const [typingTimeout, setTypingTimeout] = useState<number | undefined>(
+    undefined,
+  );
+
+  const [filterValue, setFilterValue] = useState('');
+
+  const handleSelectedPerson = (person: Person | null) => {
+    setSelectedPerson(() => person);
+    setFilterValue('');
+  };
+
+  const handleChangeText = (event: React.ChangeEvent<HTMLInputElement>) => {
+    const value = event.target.value;
```

Comment:
You can also put event.target.value straight to your function 
```suggestion
    setFilterValue(event.target.value);
```

Reply:
good point 🔥

Reply:
u don't need a separate variable for it. u used it only once.